introspection-engine/introspection-engine-tests/tests/re_introspection/vitess.rs

```diff
@@ -0,0 +1,337 @@
+          bs  B[]
+          id  Int    @id @default(autoincrement())
+        }
+
+        model B {
+          id  Int @id @default(autoincrement())
+          a   A   @relation(fields: [aId], references: [id])
+          aId Int
+        }
+    "#};
+
+    let expected = expect![[r#"
+        model A {
+          bs B[]
+          id Int @id @default(autoincrement())
+        }
+
+        model B {
+          id  Int @id @default(autoincrement())
+          a   A   @relation(fields: [aId], references: [id])
+          aId Int
+        }
+    "#]];
+
+    expected.assert_eq(&api.re_introspect_dml(input_dm).await?);
+
+    Ok(())
+}
+
+#[test_connector(tags(Vitess), preview_features("referentialIntegrity"))]
+async fn deleting_models_will_delete_relations(api: &TestApi) -> TestResult {
+    let dml = indoc! {r#"
+        CREATE TABLE `A` (
+            id INT AUTO_INCREMENT PRIMARY KEY
+        );
+
+        CREATE TABLE `B` (
+            id  INT AUTO_INCREMENT PRIMARY KEY,
+            aId INT NOT NULL
+        );
+    "#};
+
+    api.database().raw_cmd(dml).await?;
+
+    let input_dm = indoc! {r#"
+        model A {
+          id Int @id @default(autoincrement())
+          bs B[]
+          cs C[]
+        }
+
+        model B {
+          id  Int @id @default(autoincrement())
+          aId Int
+          a   A   @relation(fields: [aId], references: [id])
+        }
+
+        model C {
+          id  Int @id @default(autoincrement())
+          aId Int
+          a   A   @relation(fields: [aId], references: [id])
+        }
+    "#};
+
+    let expected = expect![[r#"
+        model A {
+          id Int @id @default(autoincrement())
+          bs B[]
+        }
+
+        model B {
+          id  Int @id @default(autoincrement())
+          aId Int
+          a   A   @relation(fields: [aId], references: [id])
+        }
+    "#]];
+
+    expected.assert_eq(&api.re_introspect_dml(input_dm).await?);
+
+    Ok(())
+}
+
+#[test_connector(tags(Vitess), preview_features("referentialIntegrity"))]
+async fn field_renames_keeps_the_relation_intact(api: &TestApi) -> TestResult {
+    let dml = indoc! {r#"
+        CREATE TABLE `A` (
+            id INT AUTO_INCREMENT PRIMARY KEY
+        );
+
+        CREATE TABLE `B` (
+            id  INT AUTO_INCREMENT PRIMARY KEY,
+            aId INT NOT NULL
+        );
+    "#};
+
+    api.database().raw_cmd(dml).await?;
+
+    let input_dm = indoc! {r#"
+        model A {
+          id Int @id @default(autoincrement())
+          bs B[]
+        }
+
+        model B {
+          id  Int @id @default(autoincrement())
+          xId Int
+          a   A   @relation(fields: [xId], references: [id])
+        }
+    "#};
+
+    let expected = expect![[r#"
+        model A {
+          id Int @id @default(autoincrement())
+          bs B[]
+        }
+
+        model B {
+          id  Int @id @default(autoincrement())
+          aId Int
+          a   A   @relation(fields: [xId], references: [id])
```

Comment:
This is invalid, is it expected? I think it's the right way round, just making sure.

Reply:
The choice here was to keep an invalid relation field in the data model, and let user fix the problems after an introspection is done. Less work than to rewrite the whole @relation attribute...